### DISCUSSION TITLE
Worker pool in Risor

### DISCUSSION THREAD

**Discussion by rubiojr** [deepnoodle-ai/risor#289] [Show and tell]
Hey all!

I've created a simple worker pool in Risor, looks something like:

```Go
import pool
// Create a pool with 2 workers
pool.new(2)

// Queue up some jobs
for range 10 {
  pool.queue(func() { time.sleep(1); print("hello") })
}

// Wait for all jobs to finish
pool.wait()
```

Code [here](https://github.com/rubiojr/risor-libs/blob/main/lib/pool.risor), improvements welcome!

**Comment by myzie** [comment]
Making good use of `spawn` eh? :-)

Looks pretty good at a glance. Related to this, I've been wondering whether https://github.com/sourcegraph/conc would be worth integrating with, or perhaps Risor's `spawn` mechanism makes it better to do something Risor specific 🤔

**Reply by rubiojr** [reply]
I haven't used conq, but looks pretty cool, I'll take a peep 👀

**Reply by rubiojr** [reply]
@myzie coincidentally, I was thinking yesterday while working the pool that Risor could benefit from some additional concurrency primitives like mutexes atomic etc
